ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -2612,13 +2676,24 @@ and perform_exn ?subtask ?result (op : operation) (t : Xenops_task.task_handle)
                       ; [VGPU_start (vgpus, true)]
                       ]
               in
+              let mem_fd =
+                match handshake with
+                | Some _ ->
+                    receive id s (* new handshake protocol *)
+                | None ->
+                    s
+                (* receiving memory on this conncection *)
+              in
               perform_atomics
                 (List.concat
                    [
-                     vgpu_start_operations
+                     vgpu_start_operations ()
```

Comment:
Why does this need to become a function now?

Reply:
There is no particular reason except I became a bit nervous that this is evaluated before it is used and given the complexity of the value would like to make sure it is constructed in the right context.